### ISSUE TITLE
invert the silicon 1082.7 nm line together with the helium line

### ISSUE THREAD

**Issue by reza35** [aasensio/hazel#16]
It is often the case that the strong Si I 1082.7 nm line is observed along with the He I 1083 nm line. The line core forms in the NLTE condition as discussed in several papers but nevertheless it is often used to measure the photosphereic magnetic field vector. By default, a Milne-Edington code cannot fit two lines together but one can images an independent atmosphere that fits the silicon line along with the main code that inverts the helium line. As the polarization signal of the silicon line is dominated by the Zeeman effect, it will not be computationally expensive compared to the Hanle computations.

**Comment by cdiazbas** [comment]
Now, a useful strategy is to use another inversion code such as SIR (more flexible than a ME code in terms of complex profiles) to model the silicon line and subtract the Stokes profiles modeled, removing the contamination to the helium profile. Other methods (such as https://arxiv.org/pdf/1604.05470.pdf) have also been proposed to address this problem. After that you can use Hazel to invert the He I 10830 line. Of course, the Silicon line is formed under NLTE conditions, but if your objective is  an estimation of the photospheric magnetic field or the subtraction of the Si line a LTE code as SIR could be a good option.

**Comment by aasensio** [comment]
In my planned rewrite of Hazel, I think it makes sense to add other (one or more) possibilities for synthesizing other lines apart from the He I multiplets. One can propose several different models, and invert their parameters simultaneously. This is taking more momentum and I'm starting to be convinced to open a new branch with this idea.

**Comment by reza35** [comment]
correct, but things can be more complicated. Imaging you want a two-component inversion: one magnetic atmosphere and one non-magnetic atmosphere (or one component plus straylight). The question is then from which inversion do you estimate the filling factor in each pixel: from the silicon run or from the helium run? or make it an iterative process? you can think about locking other parameters together like the macro-turbulence velocity. So it is helpful to run the two lines in one job.
